File: yetutil/src/main/java/net/yet/util/app/YetApp.java
package net.yet.util.app;

import android.app.Application;

import net.yet.R;
import net.yet.ui.activities.AppVisibleListener;
import net.yet.util.xlog;

/**
 * Created by yet on 2015/10/10.
 */
public class YetApp extends Application implements AppVisibleListener {

	@Override
	public void onCreate() {
		this.setTheme(R.style.LibTheme);
		super.onCreate();
		App.init(this);
	}

	@Override
	public void onEnterForeground() {
		xlog.d("onEnterForeground");
	}

	@Override
	public void onEnterBackground() {
		xlog.d("onEnterBackground");
	}
}

File: yetutil/src/main/java/net/yet/util/app/YetApp.kt
package net.yet.util.app

import android.app.Application
import net.yet.theme.TextSize
import net.yet.ui.activities.AppVisibleListener
import net.yet.util.log.xlog

/**
 * Created by yet on 2015/10/10.
 */
open class YetApp : Application(), AppVisibleListener {

	override fun onCreate() {
		this.setTheme(net.yet.R.style.yetTheme)
		super.onCreate()
		App.init(this)
		if (OS.HUAWEI) {
			TextSize.addTextSize(-1)
		}
	}

	override fun onEnterForeground() {
		xlog.d("onEnterForeground")
	}

	override fun onEnterBackground() {
		xlog.d("onEnterBackground")
	}
}
